### ISSUE TITLE
[Bug]: recyclecoach_com source broken in 2.2.0

### ISSUE THREAD

**Issue by gregprosser** [mampfes/hacs_waste_collection_schedule#2845]
### I Have A Problem With:

A specific source, The integration in general

### What's Your Problem

In the released 2.2.0 version, recyclecoach_com no longer provides valid JSON.  It looks like the fix is in your master branch, can you please cut a release so others can update?

(For now, I've installed your master branch manually to get myself working again)

It looks like this commit fixes it: https://github.com/mampfes/hacs_waste_collection_schedule/commit/d3598a2c5f9892952ff995ffbef353c1a02546ee

Alternately: can you expose the ability to install the master branch from within HACS?  I tried googling and I actually don't really understand how you'd do that, but I only see the actual set releases when I install from the HACS store. I notice it's in the checkbox (try redownload), but master isn't listed for me.

### Source (if relevant)

recyclecoach_com

### Logs

```Shell
2024-10-14 11:03:16.520 ERROR (SyncWorker_6) [custom_components.waste_collection_schedule.waste_collection_schedule.source_shell] fetch failed for source Recycle Coach:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/requests/models.py", line 974, in json
    return complexjson.loads(self.text, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/simplejson/__init__.py", line 514, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/simplejson/decoder.py", line 386, in decode
    obj, end = self.raw_decode(s)
               ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/simplejson/decoder.py", line 416, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 145, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/recyclecoach_com.py", line 196, in fetch
    self._lookup_city()
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/recyclecoach_com.py", line 120, in _lookup_city
    city_data = res.json()
                ^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/requests/models.py", line 978, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
```


### Relevant Configuration

_No response_

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
I will create a release very soon. HACS removed the easy way to install the master branch in version 2.0.0 you know need to run an update service:


install master/git: Developer tools -> actions -> select `update: Install update` -> click `Coose entity` and select WCS -> select `version` and type `master` in the input field -> run the action -> restart HA


<details><summary>Older versions of HACS:</summary>
<p>
install master with HACS version < 2.0: (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
</p>
</details>

**Comment by 5ila5** [comment]
I just released 2.3.0 which contains the fix for this
